vcf/methods.go

```diff
@@ -1,21 +1,66 @@
 package vcf
 
 import (
-	"fmt"
 	"io"
+	"strconv"
 	"strings"
 
 	"github.com/vertgenlab/gonomics/dna"
 )
 
 // String implements the fmt.Stringer interface for easy printing of Vcf with the fmt package.
 func (v Vcf) String() string {
-	return fmt.Sprintf("%s\t%v\t%s\t%s\t%s\t%v\t%s\t%s\t%s\t%s\n", v.Chr, v.Pos, v.Id, v.Ref, strings.Join(v.Alt, ","), v.Qual, v.Filter, v.Info, v.Format, SamplesToString(v.Samples))
+	var buf strings.Builder
+	buf.Grow(256) // Pre-allocate some space for efficiency.
```

Comment:
I have often thought about implementing a writer that functions similarly to the `GoReadToChan` functions except it outputs a sender channel that you can send records to for writing. The benefit is that you don't need to reallocate infrastructure for each write call and it would generally promote improved performance across the codebase as responsibility for writing is naturally delegated away from the main thread. I think with switching to `strings.Builder` the benefit of a `GoWriteFromChan` system would be even greater (though might require a specialized function that inputs the buffer). 

Anyway, nothing that needs to be changed. Just sharing some thoughts. I am a big fan of this improvement.

Reply:
I definetely thought of something along those lines. One implementation of this method was to pass in the `strings.Builder` variable, so the function could look something like:
`func (v Vcf, buf strings.Builder) String() string {}` but I wasn't sure how it would perform within goroutines. I decided that this would be a good first pass and ensure satefy. I can add a TODO: to revisit this later